app/components/hook-switch.js

```diff
@@ -13,12 +18,12 @@ export default Ember.Component.extend({
       let pusher = this.get('pusher'),
         repoId = hook.get('id');
 
-      return hook.toggle().then(() => {
+      yield hook.toggle().then(() => {
         pusher.subscribe(`repo-${repoId}`);
       }, () => {
         this.toggleProperty('hook.active');
         return this.sendAction('onToggleError', hook);
       });
     }
-  }
+  }),
```

Comment:
Of course! I was thinking for some reason that this was happening in a parent component :facepalm:. Going to merge this in :)